lib/src/rules/avoid_double_and_int_checks.dart

```diff
@@ -51,16 +51,19 @@ class AvoidDoubleAndIntChecks extends LintRule {
             group: Group.style);
 
   @override
-  AstVisitor getVisitor() => new Visitor(this);
+  void registerNodeProcessors(NodeLintRegistry registry) {
+    final visitor = new _Visitor(this);
+    registry.addIfStatement(this, visitor);
```

Comment:
I've missed a bit of the conversation but this pattern worries me a little for rule authors.  Is there always a one-to-one correlation between registry additions and visit method implementations?

EDIT: I'm more OK with it after some thinking; I'm also guessing the answer may be in general yes but in particular not always.  Documentation would be really useful here (but maybe you have some already ;))

Reply:
Yes, basically we override methods of `SimpleAstVisitor` for nodes we are interested in. And we register our intentions, so that the driving AST visitor does not waste time doing megamorphic invocations for all the other (empty, but to driving visitor cannot know this) methods. So, yes, it is one-to-one.